glommio/src/task/raw.rs

```diff
@@ -14,6 +14,9 @@ use crate::task::header::Header;
 use crate::task::state::*;
 use crate::task::utils::{abort, abort_on_panic, extend};
 use crate::task::Task;
+use std::thread::ThreadId;
+
+thread_local!(static THREAD_ID : ThreadId = std::thread::current().id());
```

Comment:
I don't understand from your comment why do you think this is beneficial as a TLS variable.

I already expect `std::thread::current()` to access a TLS variable to store its current pointer. Is that not the case?

Once you have the current thread info pointer, the id is just an u64 according to https://doc.rust-lang.org/src/std/thread/mod.rs.html#970

Creating the ID requires a bunch of atomics, but that's ok. Accessing it seems like fast.

Reply:
Hi @glommer here is the two simple benches which compare the performance of both approaches https://gist.github.com/laa/d71cb4bba211b1ecfc52bed0af3c91f6. 
In nutshell access through  std::thread::current() takes 10 ns, but access through try_with takes 0 ns. (obviously it takes some but it needs to be measure with higher precision).  I can add as the comment to the code. Sorry for not doing that it seems to be clear for me. But as I can see it is not clear enough :-)

Reply:
It is not your fault. This is really puzzling!

I wonder why this is. Both options should be essentially a TLS access with an offset.

Do you have any explanation? I am ok with merging this - a comment close to the call site would definitely help.
But I wonder which code are both versions  generating?

Reply:
AFAIK, can not look at the code right now. Thread is protected by Arc which is cloned on the access of the thread. That is heavier obviously then getting id from the TLS. I will benchmark Arc performance out of curiosity, but unlikely I will get 10 ns. overhead, but who knows. Except for that, I do not have an explanation.  Maybe call overhead not 10 ns but less in the long run, for run15 min for example. The current bench is running about 2-3 min, which IMHO still long enough. But the difference is visible at least on my server.

Reply:
I will add a comment a bit later, but interesting that I always get the same consistent results several runs. So that is not a problem of tasks which are running at the moment on the server. Which was idle actually.

Reply:
If the access to the current thread is protected by Arc that explains it.
But that's really surprising an unfortunate! =(

I'll merge as is and we can add a comment later.